### ISSUE TITLE
About binary segmentation

### ISSUE THREAD

**Issue by Jinshan99** [open-mmlab/mmsegmentation#1873]
Hello, I have a question on training the model for binary class, crack and noncrack.They returned this, 2022-08-05 12:34:38,640 - mmseg - INFO - Iter(val) [1551]	aAcc: 0.9814, mIoU: 0.8320, mAcc: 0.9205, IoU.crack: 0.9806, IoU.non-crack: 0.6833, Acc.crack: 0.9877, Acc.non-crack: 0.8533, So for the binary segmentation, I think the result should be as one accuracy, one IoU... How to solve it~~ Thank you!!

**Comment by xiexinch** [comment]
Hi @Jinshan99,
There are some bugs when setting `num_classes=1`, we're working on solving this problem. #1774

**Comment by Jinshan99** [comment]
Hi, could you show me the code of calculate precision~~, I got very large result when I  add
    label = label[mask]
    
    matt=matthews_corrcoef(label, pred_label)  
    matt=torch.tensor(matt)

Matt.crack: 17278964.4800, Matt.non-crack: 964808.0000

**Comment by xiexinch** [comment]
Hi @Jinshan99,
Might take a look at this discussion. https://github.com/open-mmlab/mmsegmentation/issues/427

**Comment by erceguder** [comment]
Hey @xiexinch,

I am working on a binary segmentation problem with a UNet-PSPNet model on a dataset created with two _distinct_ classes (`num_classes = 2`). The possible problem is that the evaluation reports two _non-equal_ accuracies, let me explain:

Bearing in mind the calculation of accuracy: (TP + TN) / (TP + TN + FP + FN), and that one class' TP is the other class' TN and vice versa I believe they should be equal in all cases. Is there any problems or suggestions with this?

Thank you in advance
